### PR TITLE
[resumes][feat] add resumes schema

### PR DESCRIPTION
**Pull request by keanecjy**
https://lucid.app/lucidchart/bcaf2307-e8df-4eba-b505-9c869e95cffc/edit?invitationId=inv_244e5678-2aad-4510-8701-22c94b320e7d&page=0_0#

### CODE REVIEW

File: apps/portal/prisma/schema.prisma [hunk 1]
```diff
@@ -76,6 +80,64 @@ enum TodoStatus {
...
+    PROJECTS
+    SKILLS
+}
+
+model ResumesStar {
+    id        String        @id @default(cuid())
+    resumeId  String
+    userId    String
+    createdAt DateTime      @default(now())
+    updatedAt DateTime      @updatedAt
```
Comment (Vielheim) [inline apps/portal/prisma/schema.prisma:112]: If the user un-stars a resume, are we just deleting the star? If so would we still need an `updatedAt` field?

File: apps/portal/prisma/schema.prisma [hunk 2]
```diff
@@ -76,6 +80,64 @@ enum TodoStatus {
...
+    user        User                 @relation(fields: [userId], references: [id], onDelete: Cascade)
+    resume      ResumesResume        @relation(fields: [resumeId], references: [id], onDelete: Cascade)
+    votes       ResumesCommentVote[]
+}
+
+model ResumesCommentVote {
+    id        String         @id @default(cuid())
+    commentId String
+    userId    String
+    value     Int
```
Comment (Vielheim) [inline apps/portal/prisma/schema.prisma:134]: Do we need to bound this value to -1 and 1?

**Comment by yangshun** [comment]
Ah crap I didn't hit submit for my review comments when I reviewed this morning. But I had very similar comments to @Vielheim

File: apps/portal/prisma/schema.prisma [hunk 3]
```diff
@@ -76,6 +80,64 @@ enum TodoStatus {
 // across all models in this file.
 // End of Resumes project models.
 
+model ResumesResume {
```
[the remark this answers is not in the document]
Comment (keanecjy) [reply apps/portal/prisma/schema.prisma:93]: Oops missed the text, thanks for pointing out!

File: apps/portal/prisma/schema.prisma [hunk 4]
```diff
@@ -76,6 +80,64 @@ enum TodoStatus {
...
+    description String               @db.Text
+    section     Section
+    createdAt   DateTime             @default(now())
+    updatedAt   DateTime             @updatedAt
+    user        User                 @relation(fields: [userId], references: [id], onDelete: Cascade)
+    resume      ResumesResume        @relation(fields: [resumeId], references: [id], onDelete: Cascade)
+    votes       ResumesCommentVote[]
+}
+
+model ResumesCommentVote {
```
[the remark this answers is not in the document]
Comment (keanecjy) [reply apps/portal/prisma/schema.prisma:137]: Yup value will be 1 and -1, will do the validations on the router level, prisma doesn't support validations yet :(

File: apps/portal/prisma/schema.prisma [hunk 5]
```diff
@@ -76,6 +80,64 @@ enum TodoStatus {
...
+    title          String           @db.Text
+    additionalInfo String           @db.Text
+    // TODO: role
+    // TODO: experience
+    // TODO: location
+    url            String
+    createdAt      DateTime         @default(now())
+    updatedAt      DateTime         @updatedAt
+    user           User             @relation(fields: [userId], references: [id], onDelete: Cascade)
+    stars          ResumesStar[]
```
[the remark this answers is not in the document]
Comment (keanecjy) [reply apps/portal/prisma/schema.prisma:103]: We discussed internally within our team and think decided to use stars to encapsulate the functionality of vote + bookmark. Stars seem to be a better representation of how popular a resume is as well

**Pull request merged**